src/garage/torch/algos/trpo.py

```diff
@@ -0,0 +1,174 @@
+        use_softplus_entropy (bool): Whether to estimate the softmax
+            distribution of the entropy to prevent the entropy from being
+            negative.
+        stop_entropy_gradient (bool): Whether to stop the entropy gradient.
+        entropy_method (str): A string from: 'max', 'regularized',
+            'no_entropy'. The type of entropy method to use. 'max' adds the
+            dense entropy to the reward for each time step. 'regularized' adds
+            the mean entropy to the surrogate objective. See
+            https://arxiv.org/abs/1805.00909 for more details.
+
+    """
+
+    def __init__(self,
+                 env_spec,
+                 policy,
+                 baseline,
+                 max_path_length=500,
+                 policy_lr=3e-4,
+                 n_samples=1,
+                 discount=0.99,
+                 gae_lambda=0.98,
+                 max_kl=0.01,
+                 center_adv=True,
+                 positive_adv=False,
+                 optimizer=None,
+                 optimizer_args=None,
+                 policy_ent_coeff=0.0,
+                 use_softplus_entropy=False,
+                 stop_entropy_gradient=False,
+                 entropy_method='no_entropy'):
+        if optimizer is None:
+            optimizer = ConjugateGradientOptimizer
+            optimizer_args = {'max_constraint_value': max_kl}
+
+        super().__init__(env_spec, policy, baseline, max_path_length,
+                         policy_lr, n_samples, discount, gae_lambda,
+                         center_adv, positive_adv, optimizer, optimizer_args,
+                         policy_ent_coeff, use_softplus_entropy,
+                         stop_entropy_gradient, entropy_method)
+
+        self._kl = None
+
+    def _compute_objective(self, advantages, valids, obs, actions, rewards):
+        """Compute the surrogate objective.
+
+        Args:
+            advantages (torch.Tensor): Expected rewards over the actions
+            valids (list[int]): length of the valid values for each path
+            obs (torch.Tensor): Observation from the environment.
+            actions (torch.Tensor): Predicted action.
+            rewards (torch.Tensor): Feedback from the environment.
+
+        Returns:
+            torch.Tensor: Calculated objective values
+
+        """
+        with torch.no_grad():
+            old_ll = self._old_policy.log_likelihood(obs, actions)
+
+        new_ll = self.policy.log_likelihood(obs, actions)
+        likelihood_ratio = (new_ll - old_ll).exp()
+
+        # Calculate surrogate
+        surrogate = likelihood_ratio * advantages
+
+        return surrogate
+
+    def _optimize(self, itr, paths, valids, obs, actions, rewards):
+        closure = self._build_closure(itr, paths, valids, obs, actions,
+                                      rewards)
+        self._optimizer.step(closure)
+
+    def _build_closure(self, itr, paths, valids, obs, actions, rewards):
```

Comment:
is there a reason you did this instead of just using lambdas?

Reply:
https://lintlyci.github.io/Flake8Rules/rules/E731.html

Reply:
i'd prefer you just `# noqa: E731` them than add 50 lines of boilerplate. also, the rule forbids assigning them, not passing them as function arguments. seems to me you could:

```python
def _optimizer(...):
    self._optimizer.step(
        compute_loss=lambda: self._compute_loss(itr, paths, valids, obs, actions, rewards)
        compute_kl=lambda: self._compute_kl_constraint(obs)
    )
```

Reply:
up to you. i think either is a better option than what you had to do here.